src/cudamatrix/cu-matrix.h

```diff
@@ -513,11 +513,22 @@ class CuMatrixBase {
   // (*this).
   void AddElements(Real alpha, const std::vector<MatrixElement<Real> >& input);
 
-  // This function resizes the output to indexes.size(), and for each element of
-  // "indexes" it interprets it as a (row, column) index into *this, and puts
-  // (*this)(row, column) into the corresponding element of "output".
+  // For each i, with indexes[i] = (j, k), does (*this)(j, k) += input[i].
+  // Requires, but does not check, that the vector of indexes does not contrain
+  // repeated elements, 'input' is the start of an array of length equal
+  // to indexes.Dim().
+  void AddElements(Real alpha, const CuArray<Int32Pair> &indexes, const Real *input);
+
+  // This function implies the output is allocated and the size is equal to
```

Comment:
changes 'implies' to 'requires'.  References to argument names should be inside single quotes (this is a new rule), so e.g. output should be 'output'.  Please make very clear that 'output' is a host vector (i.e. on the CPU).